chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -264,6 +264,23 @@ def self.unclaimed(node, include_mounted=false)
           end
         end
 
+        # return the disks that are unclaimed and prepared for given role (or with no role)
+        # (sort the result so the ones with exact role come first)
```

Comment:
My idea is that it could still work the current way, barclamps just claiming disks as they were before (semi-randomly), but you can pick some of them to use specific roles.

For example, with cinder, when you have "take first" option set, it will take the one with cinder role, because that's how sorting is done here in ``prepared_for`` function. If you want to "take all", than it would take also disks that are unassigned, but that's what you should expect when asking for this. But most importantly, it will not use any disk that is marked for anything non-cinder